webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/campaigns/MainCampaignService.scala

```diff
@@ -433,6 +433,31 @@ object MainCampaignScheduler {
           Inconsistency(s"Cannot schedule a one shot event if end (${DateFormaterService
               .getDisplayDate(end)}) date is before start date (${DateFormaterService.getDisplayDate(start)})").fail
         }
+
+      case Daily(start, end) =>
+        val startDate = (if (
+                           date.getHourOfDay() > start.realHour || (date.getHourOfDay() == start.realHour && date
+                             .getMinuteOfHour() > start.realMinute)
+                         ) {
+                           date.plusDays(1)
+                         } else {
+                           date
+                         })
+          .withHourOfDay(start.realHour)
+          .withMinuteOfHour(start.realMinute)
+          .withSecondOfMinute(0)
+          .withMillisOfSecond(0)
+
+        val endDate = {
+          (if (end.realHour < start.realHour || (end.realHour == start.realHour && end.realMinute < start.realMinute)) {
+             startDate.plusDays(1)
+           } else {
+             startDate
+           }).withHourOfDay(end.realHour).withMinuteOfHour(end.realMinute).withSecondOfMinute(0).withMillisOfSecond(0)
+        }
+
+        Some((startDate, endDate)).succeed
```

Comment:
I think we want to limit the max span to 20h or so to avoid all the borderline cases when two campaign override (ie, we don't even want to have to wonder what would happen). 

At least, it needs to be less than 24h